truffe2/main/test_tools.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+'''
+Unit test tools for truffe
+
+'''
+from bs4 import BeautifulSoup
+from json import loads
+import os
+import sys
+from _io import StringIO
+from os.path import dirname
+
+from django.test import TestCase, Client
+from django.test.client import MULTIPART_CONTENT
+from django.core.management import call_command
+
+from main.test_data import initial_data
+
+
+class TruffeTestAbstract(TestCase, Client):
+
+    def __init__(self, methodName):
+        TestCase.__init__(self, methodName)
+        Client.__init__(self)
+        self.send_content_type = MULTIPART_CONTENT
+
+    def setUp(self):
+        TestCase.setUp(self)
+        initial_data()
+        class_name = self.__class__.__name__
+        if 'WithLogin' in class_name:
```

Comment:
Not possible in the constructor of unitcase class

Reply:
Could you please develop? Can't we just do something like this?

```python3
class A:
    def __init__(self):
        pass

class B:
    def __init__(self):
        pass

class C(A, B):
    def __init__(self, *args, **kwargs):
        self.with_client = False
        if with_client in kwargs:
            self.with_client = kwargs['with_client']
            del kwargs['with_client']
        super(A, self).__init__(*args, **kwargs)
```

Otherwise, how about setting the attributes in the derived class' setUp method and adding a note in the class documentation?

Reply:
Change algorithme to select test user in "Step2"